src/components/Popup/Popup.tsx

```diff
@@ -130,12 +142,36 @@ export default class Popup extends AutoControlledComponent<ReactProps<PopupProps
     accessibility: popupBehavior,
     align: 'start',
     position: 'above',
+    on: 'click',
+    mouseLeaveDelay: 500,
   }
 
   public static autoControlledProps = ['open', 'target']
 
   private static isBrowserContext = isBrowser()
 
+  private isPopupClosing = false
+
+  setPopupOpen(newOpen, e) {
+    if (!newOpen) {
+      this.schedulePopupClose(e)
+    } else {
+      this.isPopupClosing = false
+      this.trySetOpen(true, e)
+    }
+  }
+
+  schedulePopupClose = e => {
+    const { mouseLeaveDelay } = this.props
+    this.isPopupClosing = true
+    setTimeout(() => {
```

Comment:
if `mouseLeaveDelay` === `0` maybe we don't need to use `setTimeout` at all? As it still causes some delay.

Reply:
I would disagree with this. Code should be either always sync or always async. Making it conditionally async can be surprising and difficult to debug.

Reply:
Changed back the initial logic for always setting up a timeout.

Reply:
We should clear this timeout in `componentWillUnmount()`